Automattic/jetpack
# Update coming soon badge in menu to display for coming soon v2 sites

In the wp-admin interface for both simple and atomic sites on wpcom, the (Coming Soon) badge is not shown in the admin menu for coming soon v2 sites

Expected:
![Screen Shot 2021-07-09 at 6 55 17 pm](https://user-images.githubusercontent.com/22446385/125052378-408d4780-e0e7-11eb-89b9-3b76daceee3d.png)

Actual:
![Screen Shot 2021-07-09 at 6 55 23 pm](https://user-images.githubusercontent.com/22446385/125052399-4551fb80-e0e7-11eb-862e-da826ed78a1f.png)

https://github.com/Automattic/wp-calypso/issues/54298

### What is coming soon v2
Coming soon v1 worked by making sites private by default, but it had issues with bugs in atomic sites. Coming soon v2 was an attempt to simplify things by making sites public by default but have their content hidden behind a coming soon page pbAok1-17l-p2.

Coming soon v1 uses the following flags:

```
blog_public = -1
wpcom_coming_soon = 1
```

Coming soon v2 sites have the following flags:
```
blog_public = 0
wpcom_public_coming_soon = 1
```

Coming soon v1 details: PCYsg-py7-p2
Coming soon v2 details PCYsg-u4S-p2

#### Does this pull request change what data or activity we track or use?
N/A

#### Testing instructions:

I actually haven't been able to work out how to apply this diff on wpcom simple or atomic sites.
When the diff is applied:
* Create a new site via `https://wordpress.com/new`, do not launch the site.
* go to wp-admin for the new site: `https://$NEW_SITE.wordpress.com/wp-admin`
* The site should have a (Coming Soon) badge in the admin menu.

For the Atomic site:
* Make the site atomic by upgrading to the business plan and installing a plugin.
* Go to wp-admin for the Atomic site: `https://$NEW_SITE.wpcomstaging.com/wp-admin`
* The site should have a (Coming Soon) badge in the admin menu.
